CONTRIBUTING.md

```diff
@@ -41,18 +45,19 @@ Read the [documentation](https://github.com/nginxinc/kubernetes-ingress/tree/mai
 
 ### Report a Bug
 
-To report a bug, open an issue on GitHub with the label `bug` using the available bug report issue template. Please ensure the issue has not already been reported.
+To report a bug, open an issue on GitHub and choose the type 'Bug report'. Please ensure the issue has not already been reported, and that you fill in the template as provided, as this can reduce turnaround time.
 
-### Suggest an Enhancement
+### Suggest a new feature or other improvement
 
-To suggest an enhancement, please create an issue on GitHub with the label `enhancement` using the available feature issue template.
+To suggest an new feature or other improvement, create an issue on Github and choose the type 'Feature request'. Please fill in the template as provided.
 
 ### Open a Pull Request
 
+* Before working on a possible pull request, you should first open an associated issue (feature request or bug report) describing the proposed change. This gives an opportunity for the core development team to discuss the potential pull request with you before you do the work.
```

Comment:
I don't think we should require an issue for every PR. It might be needed for some big or controversial change, but in most cases seeing the code changes actually helps. We probably should change the PR template to encourage people to give a more detailed description of the PR.

Reply:
I can see the value in that. However as we want to move towards a 'github first' workflow in general, I think all work by external contributors should start with an issue. This will allow us to apply the issue workflow (and perhaps kindly turn down the issue) before the contributor goes to the trouble of creating a PR, thus saving them work.

If we decide that it is too onerous, we can always implement the changed PR template at that point.

Reply:
we can apply the issue workflow to PRs as well